simpleflow/canvas.py

```diff
@@ -108,24 +109,58 @@ class Group(object):
     def __init__(self,
                  *activities,
                  **options):
-        self.activities = list(activities)
+        self.activities = []
         self.max_parallel = options.pop('max_parallel', None)
         self.raises_on_failure = options.pop('raises_on_failure', None)
-        if self.raises_on_failure is not None:
-            for act in activities:
-                propagate_attribute(act, 'raises_on_failure', self.raises_on_failure)
+        self.extend(activities)
 
-    def append(self, *args, **kwargs):
-        if isinstance(args[0], (Submittable, Group)):
+    def append(self, submittable, *args, **kwargs):
+        if isinstance(submittable, (Submittable, Group)):
+            if args or kwargs:
+                raise ValueError('args, kwargs not supported for Submittable or Group')
             if self.raises_on_failure is not None:
-                propagate_attribute(args[0], 'raises_on_failure', self.raises_on_failure)
-            self.activities.append(args[0])
-        elif isinstance(args[0], Activity):
+                propagate_attribute(submittable, 'raises_on_failure', self.raises_on_failure)
+            self.activities.append(submittable)
+        elif isinstance(submittable, Activity):
             if self.raises_on_failure is not None:
-                propagate_attribute(args[0], 'raises_on_failure', self.raises_on_failure)
-            self.activities.append(ActivityTask(*args, **kwargs))
+                propagate_attribute(submittable, 'raises_on_failure', self.raises_on_failure)
+            self.activities.append(ActivityTask(submittable, *args, **kwargs))
         else:
-            raise ValueError('{} should be a Submittable, Group, or Activity'.format(args[0]))
+            raise ValueError('{} should be a Submittable, Group, or Activity'.format(submittable))
+
+    def extend(self, iterable):
+        """
+        Append the specified activities.
+        :param iterable: list of Submittables/Groups/tuples
+        Tuples are (activity, [args, [kwargs]]).
+        """
+        for it in iterable:
```

Comment:
The tuple needs unpacking:
```python
        for it in iterable:
            if not isinstance(it, tuple):
                self.append(it)
            else:
                self.append(*it)
```
And passing kwargs doesn't work anymore. Well, that was not very useful anyway, so let's simplify this way